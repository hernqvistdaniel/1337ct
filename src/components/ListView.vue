<script setup lang="ts">
import type { Presentation } from '@/models/presentation';

defineProps<{
  presentation: Presentation
}>()
</script>

<template>
  <li>
    <div class="texts">
      <p v-if="presentation?.name" class="cardText">{{ presentation.name }}</p>
      <p v-if="presentation?.office" class="cardText">Office: {{ presentation.office }}</p>

      <a v-if="presentation?.linkedIn" :href="'http://linkedin.com/' + presentation.linkedIn">
        <font-awesome-icon class="icon" :icon="['fas', 'phone']" />
      </a>
      <a v-if="presentation?.gitHub" :href="'http://github.com/' + presentation.gitHub">
        <font-awesome-icon class="icon" :icon="['fas', 'gamepad']" />
      </a>
      <a v-if="presentation?.twitter" :href="'http://twitter.com/' + presentation.twitter">
        <font-awesome-icon class="icon" :icon="['fas', 'satellite']" />
      </a>
    </div>
  </li>
</template>

<style scoped>
.texts {
  display: flex;
  flex-direction: row;
}

.iconContainer {
  display: flex;
  flex-direction: row;
}

.icon {
  padding: 2px;
  text-decoration: none;
  color: black;
  cursor: pointer;
  transition: all .2s ease-in-out;
}

.icon:hover {
  transform: scale(1.2);
}

.icon:visited {
  text-decoration: none;
  color: black;
  cursor: pointer;
}

.cardText {
  font-size: 12px;
  font-weight: 600;
  text-align: left;
}

.cardImg {
  margin: auto;
  max-height: 120px;
  align-self: center;
}
</style>
