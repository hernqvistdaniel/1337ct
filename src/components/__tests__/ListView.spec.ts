import { describe, it, expect } from 'vitest'

import { shallowMount } from '@vue/test-utils'
import ListView from '../ListView.vue'

describe('ListView.vue', () => {
  const wrapper = shallowMount(ListView, { props: {
    presentation:
      {
        "name": "Akan Murat Cimen",
        "email": "akan.murat.cimen@1337.tech",
        "phoneNumber": "076-1337198",
        "office": "Lund",
        "manager": "jim@1337.tech",
        "orgUnit": "/Employees",
        "mainText": "<p>It has always been appealing to me to try to understand everything from the big bang to AI-suggested cat videos by asking the what, why, and how questions. If I ask the right questions at the right time, in the right order, I believe everything has an answer. The knowledge we have learned as human beings so far by asking these questions is an invaluable treasure.</p><p>I have been demanding some operations from CPUs since I was a kid. I worked on various projects like e-commerce, content management systems, car sharing, live streaming etc. with various programming languages. I think the true \"legacy code\" is not about how old it is, it's about its readability, maintainability etc. I'm trying not to be that person who writes code like that.</p><p>In my spare time I'm demanding more operations from CPUs by playing games, watching movies, tv-series or traveling with the help of a navigation application.</p> ",
        "gitHub": "akanmuratcimen",
        "twitter": null,
        "stackOverflow": "513234",
        "linkedIn": "/in/akanmuratcimen/",
        "imagePortraitUrl": "https://i.1337co.de/profile/akan-murat-cimen",
        "imageWallOfLeetUrl": "https://i.1337co.de/wallofleet/akan-murat-cimen",
        "highlighted": false,
        "published": true
      } 
}})

  it('renders properly', () => {
    expect(wrapper.html()).toBeTruthy()
  })

  it('renders a font-awesome component', () => {
    const fontAwesome = wrapper.find('font-awesome-icon')
    expect(fontAwesome.html()).toBeTruthy()
  })

  it('matches a snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})