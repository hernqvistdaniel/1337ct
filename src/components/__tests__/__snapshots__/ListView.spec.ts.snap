// Vitest Snapshot v1

exports[`ListView.vue > matches a snapshot 1`] = `
"<li data-v-a09dd1da=\\"\\">
  <div class=\\"texts\\" data-v-a09dd1da=\\"\\">
    <p class=\\"cardText\\" data-v-a09dd1da=\\"\\">Akan Murat Cimen</p>
    <p class=\\"cardText\\" data-v-a09dd1da=\\"\\">Office: Lund</p><a href=\\"http://linkedin.com//in/akanmuratcimen/\\" data-v-a09dd1da=\\"\\">
      <font-awesome-icon class=\\"icon\\" icon=\\"fas,phone\\" data-v-a09dd1da=\\"\\"></font-awesome-icon>
    </a><a href=\\"http://github.com/akanmuratcimen\\" data-v-a09dd1da=\\"\\">
      <font-awesome-icon class=\\"icon\\" icon=\\"fas,gamepad\\" data-v-a09dd1da=\\"\\"></font-awesome-icon>
    </a>
    <!--v-if-->
  </div>
</li>"
`;
